website/content/docs/internals/integrated-storage.mdx

```diff
@@ -138,48 +138,98 @@ forwarded to the leader.
 Similar to other storage backends, data that is written to the Raft log and FSM
 will be encrypted by Vault's barrier.
 
+### Configurable Raft Log Store
+
+By default, Vault uses BoltDB, via the raft-boltdb library, to store raft logs. Starting in 1.16.0, Vault can
+be configured to use [raft-wal](https://github.com/hashicorp/raft-wal) instead.
+
+When Vault is configured to use BoltDB, the raft logs are stored in a file called
+`raft.db`, located inside the `raft` subdirectory of Vault's main storage directory.
+When configured to use raft-wal, Vault will instead store logs inside a `wal`
+subdirectory, again located inside the `raft` subdirectory.
+
+Why would someone want to enable raft-wal for raft's log store instead of using the
+default of BoltDB? There are a few reasons.
+
+First, consider BoltDB's freelist. BoltDB is a single file database. When data is
+deleted, it's not possible to shrink the file on disk to recover the deleted space.
+Instead, BoltDB marks the pages that the data was stored on as free for use. This
+means in the future, when writes happen and new pages need to be allocated, BoltDB
+will first consult the freelist and attempt to reuse old pages before allocating
+new ones. Normally, this works reasonably well, but if you're operating a cluster
+with high churn, e.g. lots of deletes, the freelist can grow large over time.
+The larger the freelist grows, the slower BoltDB becomes.
+
+raft-wal is designed specifically for storing raft logs, and does not use a freelist.
+raft-wal maintains a directory of files as its data store and is able to compact
+data over time to free up space when it's no longer needed.
+
+Second, consider a particular failure mode for busy Vault clusters. In some cases,
+a fresh follower joining an existing raft cluster (or an existing follower that's
+rebooted) is never able to fully join the cluster because the time it takes them
+to install a raft snapshot is great enough that by the time it's finished, the leader
+has already truncated its existing logs and needs to send the follower a new snapshot.
+In the pathological case, this repeats indefinitely, with the follower never able to
+join.
+
+The straightforward solution to this problem is to increase the number of logs that
+leaders retain before they truncate their log store. However, setting this too high
+causes the BoltDB file to grow large and be more susceptible to sudden bursts of writes,
+causing a large freelist and subsequent poor write performance. Operators have to
+carefully tune this value to be large enough for their future growth in write throughput
+and snapshot size, but no larger than necessary, as it increases the risk of poor
+performance.
+
+When using raft-wal, this number can be increased without risking poor performance
+from explosive freelist growth.
```

Comment:
Back when I worked on the enterprise autopilot features (which Consul already made use of), I tried to lean hard on the Consul docs and received a lot of pushback on that approach, so I've tried to avoid it since then. But I will take a look and see what all I can bring over.